### PR TITLE
Tests for knitting all report types, via enabling non-interactive use of create_visc_project(), use_visc_report(), and use_visc_methods()

### PR DESCRIPTION
**Pull request by slager**
This PR sets up an automated unit test for successfully knitting a generic report PDF across all platforms. This will help with future unit test writing, and gets us a lot farther along on #136. The test is also the reprex for #159, so it will greatly help me iterate on debugging that on statsrv once I merge these changes into the #159 branch.

The CI runs on this PR are failing, but for a very good reason: They're successfully catching the 'Lonely item' latex bug across all platforms. See #169 (but don't merge it!) for a demonstration that once I combine these with Kellie's bugfix, the generic reports successfully knit on all the CI platforms.

Along the way, I had to make a few changes to a few functions to enable non-interactive use (see https://github.com/FredHutch/VISCtemplates/issues/136#issuecomment-2176681161).

Changes in this PR:

- Exported functions `create_visc_project()`, `use_visc_report()`, and `use_visc_methods()` now have an additional argument `interactive` that is always `TRUE` by default, with the same behavior and UI as before. Setting `interactive = FALSE` produces the same results as before, but without any interactive side effects like automatic Rstudio window/project switching, interactive user confirmations, interactive user questions, or console outputs. The new arguments have been documented for the 3 functions.
- Internal helper functions `challenge_visc_name()`, `challenge_directory()`, and `challenge_visc_report()` also now have an interactive option that defaults to `TRUE` but can be set to `FALSE` with the same effect. These helpers are called from within the 3 exported functions above.
- Replaced use of the `here` package with `rprojroot` to facilitate unit testing.  `here` wasn't flexible enough to handle test knitting an Rmd file in a project package from within the VISCtemplates unit tests.
- CI tweaks to facilitate installing dependencies and building the report on the R CMD check runners
- Automated unit test for knitting a generic report

### CODE REVIEW

**Comment by slager** [comment]
Thought of a test I want to implement for this, so un-requested reviews for now.  Feel free to take a look but I will re-request reviews once I'm finished with that.

**Comment by slager** [comment]
This is ready for review now!

File: tests/testthat/test-template.R [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  create_visc_project(foo, interactive = FALSE)
+  # temporarily setwd() during test_that block
+  withr::local_dir(foo)
+  use_visc_report(report_type = "generic", interactive = FALSE)
+  expect_no_error(
+    rmarkdown::render(file.path('PT-Report', 'PT-Report.Rmd'), quiet = TRUE)
+  )
+  expect_true(
+    file.exists(file.path("PT-Report", "PT-Report.pdf"))
+  )
```
Comment (kelliemac) [inline tests/testthat/test-template.R:13]: Could we also add an expect_snapshot_file() call (see https://testthat.r-lib.org/articles/snapshotting.html#whole-file-snapshotting)? It looks like we can use this to save the pdf, and then it will automatically be uploaded as a GitHub artifact during the CI checks [here](https://github.com/FredHutch/VISCtemplates/actions/runs/9618562082/job/26532718862#step:8:169) so that the pdf can be reviewed on GitHub which would be really nice!

File: tests/testthat/test-template.R [hunk 2]
```diff
@@ -0,0 +1,14 @@
+test_that("generic report knits", {
```
Comment (kelliemac) [inline tests/testthat/test-template.R:1]: Can we add a test for knitting the empty report as well? And maybe bama and nab too, unless there are issues with those?

File: tests/testthat/test-template.R [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  foo <- withr::local_tempdir()
+  create_visc_project(foo, interactive = FALSE)
+  # temporarily setwd() during test_that block
+  withr::local_dir(foo)
+  use_visc_report(report_type = "generic", interactive = FALSE)
+  expect_no_error(
+    rmarkdown::render(file.path('PT-Report', 'PT-Report.Rmd'), quiet = TRUE)
+  )
+  expect_true(
+    file.exists(file.path("PT-Report", "PT-Report.pdf"))
```
Comment (kelliemac) [inline tests/testthat/test-template.R:12]: It would be good to also test knitting to the word doc output. But if this doesn't work, let's just note it as the next issue to tackle.

**Comment by kelliemac** [comment]
This looks great!! Thanks so much for doing this, Dave!! I added a few comments to tests/testthat/test-template.R about ways to make the testing a little more comprehensive. I think we may as well try to tackle these now and hopefully it won't be too much more work, but if there are any roadblocks let us know and we can push those off for later.

**Comment by slager** [comment]
Thanks! I had already started working locally on knitting the empty report also (no problems!) and the word versions, so will add those, will try adding bama/nab, and will see if I can get the snapshot artifact upload to work as well.

**Comment by kelliemac** [comment]
great! I'll keep an eye out for updates.

**Review by kelliemac — APPROVED** [review]

**Comment by slager** [comment]
Yup! It's almost there, and the snapshots are getting uploaded like you say!  The only thing I'm still looking into is having just the snapshots uploaded in the artifact, not the whole results tree.  There's either something I'm not fully understanding about the r-lib/actions yaml or a bug in how that works when tests fail. Hopefully more TBA soon, thanks for approving but let's hold off on merging for now.

**Comment by kelliemac** [comment]
sounds good!!

**Comment by slager** [comment]
I'm pretty convinced now that this is happening because of the upstream yml.  We'll see if they're amenable to a PR that would let us have the flexibility to always upload snapshots, even when tests fail. They already have that functionality built into the results artifact upload, just not for snapshots.

https://github.com/r-lib/actions/issues/871#issuecomment-2195460751

**Comment by kelliemac** [comment]
sounds good, Dave! is it worth merging this PR while we wait on the r-lib folks? or is there anything you want to wait on?

**Comment by slager** [comment]
Yep! I'll add a few more code comments today, then merge it, then add an issue/reminder for tweaking the CI if/when upstream changes their yml.

**Comment by slager** [comment]
Leaving the branch here for now in case I need to refer to it during yml discussion with r-lib/actions upstream.

**Comment by slager** [comment]
Deleting branch since there hasn't been movement on this from upstream. Will resurrect later as needed.
